src/JobExecutor.ts

```diff
@@ -108,145 +105,159 @@ export class JobExecutor implements JobApplication {
         if (this.isIdle) {
             const jobRepo = this._databaseService.getJobRepository();
             const job = await jobRepo.findOne({jobMessageId: msg.id});
-            console.log(msg.id + ' message received');
-            if (job) {
-                job.status = JobStatus.Running;
-                job.startTime = new Date();
-                await jobRepo.save(job);
-                this.processJob(job).then(() => console.log('job processed'), error => console.error(error));
+            logger.info(msg.id + ' message received');
+            if (job && job.status === JobStatus.Queueing) {
+                let resume = false;
+                if (job.jobExecutorId && job.jobExecutorId === this.id) {
+                    // this job is paused previously and ran on this JobExecutor instance
+                    resume = true;
+                } else if (job.jobExecutorId && job.jobExecutorId !== this.id) {
+                    // this job is paused previously but ran on another JobExecutor instance
+                    await sleep(1000);
```

Comment:
no, just a delay before put back to queue to avoid flood messages when no consumer is available. I have add delay in RabbitMQService, I think I can remove the sleep()

Reply:
removed this sleep() as the RabbitMQService already adds 1s delay for NACK